### PR TITLE
Add requestMethod Parameter to send request - Closes #240

### LINKED ISSUE #240: Add first parameter GET/POST/PUT for sendRequest
Part of #139

### PR DESCRIPTION
**Pull request by Tosch110**
Merge #252 first - Closes #240

### CODE REVIEW

File: src/api/privateApi.js [hunk 1]
```diff
@@ -186,113 +186,56 @@ function serialiseHttpData(data) {
...
 
-function changeRequest(requestType, options) {
-	const defaultRequestObject = {
-		requestMethod: '',
-		requestUrl: '',
-		nethash: '',
-		requestParams: '',
+	return {
+		method: requestMethod,
+		url: requestUrl,
```
Comment (willclarktech) [inline src/api/privateApi.js]: If you call these `method` and `url` in the first place you can use object shorthand here.

File: src/api/privateApi.js [hunk 2]
```diff
@@ -186,113 +186,56 @@ function serialiseHttpData(data) {
...
-	if (checkRequest.call(this, requestType, options) !== 'NOACTION') {
-		const requestValues = changeRequest.call(this, requestType, options);
-		return doPopsicleRequest.call(this, requestValues);
-	}
-	return new Promise(((resolve) => {
-		resolve({ done: 'done' });
-	}));
+function sendRequestPromise(requestMethod, requestType, options) {
+	return doPopsicleRequest.call(this, requestMethod, requestType, options);
 }
```
Comment (willclarktech) [inline src/api/privateApi.js]: Let's get rid of either this function or `doPopsicleRequest`.

File: src/api/privateApi.js [hunk 3]
```diff
@@ -186,113 +186,56 @@ function serialiseHttpData(data) {
...
-function changeRequest(requestType, options) {
-	const defaultRequestObject = {
-		requestMethod: '',
-		requestUrl: '',
-		nethash: '',
-		requestParams: '',
+	return {
+		method: requestMethod,
+		url: requestUrl,
+		headers: this.nethash,
```
Comment (willclarktech) [inline src/api/privateApi.js]: Consider passing `nethash` down as an argument to make as many functions pure as possible.

File: test/api/liskApi.js [hunk 4]
```diff
@@ -1143,4 +925,48 @@ describe('Lisk.api()', () => {
 			});
 		});
 	});
+
+	describe('#createRequestObject', () => {
+		const options = { limit: 5, offset: 3, details: 'moredetails' };
+		const LSKAPI = liskApi({ node: 'localhost' });
```
Comment (willclarktech) [inline test/api/liskApi.js]: Prefer creating fresh instances `beforeEach`. (Also for `options`.)

File: test/api/liskApi.js [hunk 5]
```diff
@@ -1143,4 +925,48 @@ describe('Lisk.api()', () => {
...
+					version: '1.0.0',
+					minVersion: '>=0.5.0',
+					port: 8000,
+				},
+				body: { limit: 5, offset: 3, details: 'moredetails' },
+			};
+
+			(requestObject).should.be.eql(expectedObject);
+		});
+	});
```
Comment (willclarktech) [inline test/api/liskApi.js]: What about
- [ ] no options object provided
- [ ] unsupported `requestType`
- [ ] empty string `requestType`
- [ ] invalid `requestMethod`
- [ ] called on object with no nethash

File: src/api/liskApi.js [hunk 6]
```diff
@@ -255,7 +251,7 @@ LiskAPI.prototype.getAddressFromSecret = function getAddressFromSecret(secret) {
  */
 
 LiskAPI.prototype.getAccount = function getAccount(address, callback) {
-	return this.sendRequest('accounts', { address }, result => callback(result));
+	return this.sendRequest('GET', 'accounts', { address }, result => callback(result));
```
Comment (Tosch110) [inline src/api/liskApi.js]: That is done in the next PR https://github.com/LiskHQ/lisk-js/pull/263/files

File: src/api/privateApi.js (hunk 3 above)
Comment (Tosch110) [inline src/api/privateApi.js]: I understand the idea and would support it, where do you have in mind nethash should be passed in first time?

Comment (willclarktech) [inline src/api/privateApi.js]: Discussed -> A job for #219

File: test/api/liskApi.js [hunk 7]
```diff
@@ -10,6 +10,8 @@ describe('Lisk.api()', () => {
 	const localNode = 'localhost';
 	const externalNode = 'external';
 	const defaultSecret = 'secret';
+	const GET = 'GET';
+	const POST = 'POST';
```
Comment (willclarktech) [inline test/api/liskApi.js]: Actually it would be nice to have these in source too.

File: test/api/liskApi.js [hunk 8]
```diff
@@ -1099,4 +927,93 @@ describe('Lisk.api()', () => {
...
+				headers: {
+					'Content-Type': 'application/json',
+					nethash: 'ed14889723f24ecc54871d058d98ce91ff2f973192075c0155ba2b7b70ad2511',
+					broadhash: 'ed14889723f24ecc54871d058d98ce91ff2f973192075c0155ba2b7b70ad2511',
+					os: 'lisk-js-api',
+					version: '1.0.0',
+					minVersion: '>=0.5.0',
+					port: 8000,
+				},
+				body: {},
```
Comment (willclarktech) [inline test/api/liskApi.js]: Now that we have several tests I would rather have a default `expectedObject` which is extended/overridden as necessary.

**Pull request merged**